src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs

```diff
@@ -103,119 +101,44 @@ public void Do()
                         return Model.CreateModel(
                             _disconnectedBufferGraph,
                             trackingSpan,
-                            totalItems,
-                            selectedItem: totalItems.First(),
+                            completionList.Items.ToList(),
+                            selectedItem: completionList.Items.First(),
                             isHardSelection: false,
                             isUnique: false,
                             useSuggestionCompletionMode: _includeBuilder,
-                            builder: _builder,
+                            builder: completionList.Builder,
                             triggerInfo: _triggerInfo,
                             completionService: _completionService,
                             workspace: _documentOpt != null ? _documentOpt.Project.Solution.Workspace : null);
                     }
                 }
 
-                private async Task<IEnumerable<CompletionItemGroup>> GetGroupsAsync(ICompletionService completionService, CompletionTriggerInfo triggerInfo, CancellationToken cancellationToken)
+                private async Task<CompletionList> GetCompletionListAsync(ICompletionService completionService, CompletionTriggerInfo triggerInfo, CancellationToken cancellationToken)
                 {
                     if (_documentOpt == null && completionService is ITextCompletionService)
                     {
                         var textCompletionService = (ITextCompletionService)completionService;
-                        return await textCompletionService.GetGroupsAsync(_text, _subjectBufferCaretPosition, triggerInfo, _completionProviders, _options, cancellationToken).ConfigureAwait(false);
+                        return await textCompletionService.GetCompletionListAsync(_text, _subjectBufferCaretPosition, triggerInfo, _completionProviders, _options, cancellationToken).ConfigureAwait(false);
                     }
                     else if (_documentOpt != null)
                     {
-                        return await completionService.GetGroupsAsync(_documentOpt, _subjectBufferCaretPosition, triggerInfo, _completionProviders, cancellationToken).ConfigureAwait(false);
+                        return await completionService.GetCompletionListAsync(_documentOpt, _subjectBufferCaretPosition, triggerInfo, _completionProviders, cancellationToken).ConfigureAwait(false);
                     }
                     else
                     {
                         return null;
                     }
                 }
-
-                private void AddGroupToMap(CompletionItemGroup group)
-                {
-                    if (group != null)
-                    {
-                        foreach (var item in group.Items.WhereNotNull())
```

Comment:
It's impossible for an item to be null at this point. There's a null check in the CompletionListContext.